elastalert/test_rule.py

```diff
@@ -4,18 +4,31 @@
 from __future__ import print_function
 
 import datetime
+import random
+import re
+import string
 
 import argparse
+import mock
+import simplejson
 import yaml
 
+from elastalert.config import load_modules
 from elastalert.config import load_options
 from elastalert.elastalert import ElastAlerter
 from elastalert.elastalert import Elasticsearch
 from elastalert.util import lookup_es_key
+from elastalert.util import replace_hits_ts
 from elastalert.util import ts_now
+from elastalert.util import ts_to_dt
+
+
+data = []
```

Comment:
Instead of making this a global, let's promote this and all of the functions that follow to a class and make `data` a dynamic member of said class.